### DISCUSSION TITLE
HELP: Get more info on the songs in a playlist

### DISCUSSION THREAD

**Discussion by xenodirt** [eritislami/evobot#268] [Q&A]
On playlist.js:
```
videos.forEach((video) => {
      song = {
        title: video.title,
        url: video.url,
        duration: video.durationSeconds
      };
```
Does anyone know how to get the thumbnail and channel name on a video in a playlist?

**Comment by brandonst** [comment]
To get the channel you would do `video.channel.title` and for the thumbnail, it's a little more different. The thumbnails come in a variety of types, default, medium, high, standard, and maxres. The resolution depends on the video, so maxres or standard may be missing, so the solution is to get the last thumbnail from the object. Here's what all this would look like:
```js
song = {
    title: video.title,
    url: video.url,
    duration: video.durationSeconds,
    thumbnail: Object.values(video.thumbnails).pop().url,
    author: video.channel.title
};
```
for a explanation on thumbnail, it gets the values from the object with `Object.values()`, then `.pop()` will get the last value, which you then can get the URL from

**Reply by xenodirt** [reply]
> To get the channel you would do `video.channel.title` and for the thumbnail, it's a little more different. The thumbnails come in a variety of types, default, medium, high, standard, and maxres. The resolution depends on the video, so maxres or standard may be missing, so the solution is to get the last thumbnail from the object. Here's what all this would look like:
> 
> ```js
> song = {
>     title: video.title,
>     url: video.url,
>     duration: video.durationSeconds,
>     thumbnail: Object.values(video.thumbnails).pop().url,
>     author: video.channel.title
> };
> ```
> 
> for a explanation on thumbnail, it gets the values from the object with `Object.values()`, then `.pop()` will get the last value, which you then can get the URL from

The duration is formatted all weird. I want it as hh:mm:ss. Also with the author, it's showing the username of the playlist creator. I want it to show the uploader of the video.

**Reply by kasepta** [reply]
use this `${new Date(song.duration * 1000).toISOString().substr(11,8)}`

but when play the playlist, duratian change to 59:59:59
idk how to fix that

**Reply by xenodirt** [reply]
> To get the channel you would do `video.channel.title` and for the thumbnail, it's a little more different. The thumbnails come in a variety of types, default, medium, high, standard, and maxres. The resolution depends on the video, so maxres or standard may be missing, so the solution is to get the last thumbnail from the object. Here's what all this would look like:
> 
> ```js
> song = {
>     title: video.title,
>     url: video.url,
>     duration: video.durationSeconds,
>     thumbnail: Object.values(video.thumbnails).pop().url,
>     author: video.channel.title
> };
> ```
> 
> for a explanation on thumbnail, it gets the values from the object with `Object.values()`, then `.pop()` will get the last value, which you then can get the URL from

The duration for the playlists only show "-1".
